### ISSUE TITLE
[Suggestion] If installed server-sided, using QuickSpyglasser will show the Player holding a spyglass

### ISSUE THREAD

**Issue by ghost** [Gluton-Official/QuickSpyglasser#2]
Would this be possible? I reckon it's not possible to force the animation through the client if you're not holding the spyglass and using the hotkey through QuickSpyglasser, but could we do this if it were installed server-sided? I'd like it if people on my server using this mod could still be seen holding a spyglass when using this mod.

**Comment by Gluton-Official** [comment]
Yes, that is one of my future plans, but figuring out which hand the spyglass should be held in when used requires a few design decisions:
- If you're using a bow, for example, both of your hands would be doing something, so should there be no spyglass animation or just the spyglass model glued to your eye with your hands not touching it?
- If both of your hands have items, should the item stay in your hand for the spyglass animation or should it not be rendered?

I've started to work on it some, but decided it wasn't necessary for the initial release and was somewhat confusing, so I'll need a bit of time to think through it. I'm currently working on the config options and then was going to add Curio support, but since that's both a client and server side feature, I'll try and see if I can do global spyglass animations client side first and if not, do the Curio stuff then server side animations.

**Comment by ghost** [comment]
> If you're using a bow, for example, both of your hands would be doing something, so should there be no spyglass animation or just the spyglass model glued to your eye with your hands not touching it?

For these actions, like boat and bow, I reckon the animation shouldn't play.

> If both of your hands have items, should the item stay in your hand for the spyglass animation or should it not be rendered?

What if playing the animation put the Spyglass in your left hand and made whatever you were holding invisible? Or maybe you could add a config for whitelisting certain items from disappearing when playing the animation, but otherwise they'd disappear?

**Comment by Gluton-Official** [comment]
Those both sound like the most reasonable solutions, and I like the whitelist/blacklist idea for cases like a glove item.
